packages/react-components/react-motion/library/src/factories/createPresenceComponent.ts

```diff
@@ -77,6 +78,8 @@ export type PresenceComponentProps = {
 export type PresenceComponent<MotionParams extends Record<string, MotionParam> = {}> = {
   (props: PresenceComponentProps & MotionParams): React.ReactElement | null;
   [MOTION_DEFINITION]: PresenceMotionFn<MotionParams>;
+  In: React.FC<MotionComponentProps & MotionParams>;
+  Out: React.FC<MotionComponentProps & MotionParams>;
```

Comment:
nit, let's re-use definitions

```suggestion
  In: ReturnType<typeof createMotionComponent<MotionParams>>;
  Out: ReturnType<typeof createMotionComponent<MotionParams>>;
```

Reply:
@layershifter Unfortunately using `ReturnType` with generic functions doesn't work in TypeScript v3.9, so this fails CI. This ability was added 3 years ago in TS v4.7, but we are using TS from 5 years ago.

See: https://stackoverflow.com/a/71560711